central/processindicator/datastore/datastore_impl.go

```diff
@@ -126,6 +130,60 @@ func (ds *datastoreImpl) removeIndicators(ctx context.Context, ids []string) err
 	return nil
 }
 
+func (ds *datastoreImpl) PruneProcessIndicators(ctx context.Context, ids []string) (int, error) {
+	if ok, err := deploymentExtensionSAC.WriteAllowed(ctx); err != nil {
+		return 0, err
+	} else if !ok {
+		return 0, sac.ErrResourceAccessDenied
+	}
+
+	return ds.pruneIndicators(ctx, ids), nil
+}
+
+func (ds *datastoreImpl) pruneIndicators(ctx context.Context, ids []string) int {
+	// Previously this used removeIndicators and would call "DeleteMany".  The issue
+	// with that is "DeleteMany" wraps the entire delete into a transaction making it an
+	// all or nothing proposition.  For pruning, if a batch fails it shouldn't fail them all.
+	// A pruning batch that fails to delete would get deleted the next iteration of pruning.
+	// So for pruning, a delete by query will be used and the IDs will be batched.  Failed
+	// batches will be logged and we will move on to the next batch.
```

Comment:
How often did the transaction failed in production environments?
If we know, why did they failed? Could it be that we have corrupted data?

Reply:
They were failing because of timeouts.  The root cause was the database was undersized and could not keep up.  These requests stacking up did not help that situation.  But things got so stacked up that when the DB size was increased it could not catch up without manual intervention.  So the idea here is to help with both situations. Buy more time before the database capacity is reached as well as allow it to recover on its own once the size is increased.